### PR TITLE
MMP-309 Validate plugin compatibility between apps and domains

### PR DESCRIPTION
**Pull request by WilliamPinho**

### CODE REVIEW

File: mule-packager/src/test/java/org/mule/tools/api/validation/resolver/visitor/MulePluginResolverVisitorTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import static org.mockito.Mockito.*;
+import static util.ResolverTestHelper.buildVisitorSpies;
+import static util.ResolverTestHelper.createProjectDependencyNodeSpy;
+
+public class MulePluginResolverVisitorTest {
+
+  private MulePluginResolverVisitor visitor;
+  private ProjectDependencyNode nodeSpy;
+  private List<DependencyNodeVisitor> visitorSpies;
+  private int NUMBER_VISITORS = 10;
```
Comment (damiansima) [inline mule-packager/src/test/java/org/mule/tools/api/validation/resolver/visitor/MulePluginResolverVisitorTest.java]: static final or lower case

File: mule-packager/src/main/java/org/mule/tools/api/validation/resolver/model/DependenciesFilter.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  /**
+   * A predicate for {@link ArtifactCoordinates} that returns true if the instance have the specific {@param scope},
+   * {@param classifier} and {@param type}.
+   * 
+   * @param classifier The classifier constraint
+   * @param scope The scope constraint
+   * @param type The type constraint
+   * @return true if the {@code ArtifactCoordinates} satisfies the constraint, false otherwise
+   */
+  protected Predicate<ArtifactCoordinates> satisfies(String classifier, String scope, String type) {
```
Comment (damiansima) [inline mule-packager/src/main/java/org/mule/tools/api/validation/resolver/model/DependenciesFilter.java]: nice!!!

File: mule-packager/src/test/java/org/mule/tools/api/validation/resolver/visitor/MulePluginResolverVisitorTest.java (hunk 1 above)
Comment (WilliamPinho) [reply mule-packager/src/test/java/org/mule/tools/api/validation/resolver/visitor/MulePluginResolverVisitorTest.java]: Done!

File: mule-packager/src/main/java/org/mule/tools/api/validation/resolver/model/DependenciesFilter.java (hunk 2 above)
Comment (WilliamPinho) [reply mule-packager/src/main/java/org/mule/tools/api/validation/resolver/model/DependenciesFilter.java]: Thank you :)

**Pull request merged**